server/form-pages/apply/risk-and-need-factors/access-and-healthcare/accessNeedsFurtherQuestions.ts

```diff
@@ -0,0 +1,129 @@
+import AccessNeeds from './accessNeeds'
+
+export type AccessNeedsFurtherQuestionsBody = {
+  needsWheelchair: YesOrNo
+  mobilityNeeds: string
+  visualImpairment: string
+  isPersonPregnant?: YesOrNo
+  otherPregnancyConsiderations: string
+  childRemoved?: YesOrNo | 'pending'
+} & ObjectWithDateParts<'expectedDeliveryDate'>
+
+@Page({
+  name: 'access-needs-further-questions',
+  bodyProperties: [
+    'needsWheelchair',
+    'mobilityNeeds',
+    'visualImpairment',
+    'isPersonPregnant',
+    'childRemoved',
+    'expectedDeliveryDate',
+    'expectedDeliveryDate-year',
+    'expectedDeliveryDate-month',
+    'expectedDeliveryDate-day',
+    'otherPregnancyConsiderations',
+  ],
+})
+export default class AccessNeedsFurtherQuestions implements TasklistPage {
+  title = 'Access, cultural and healthcare needs'
+
+  questions = {
+    wheelchair: `Does ${this.application.person.name} require a wheelchair accessible room?`,
+    mobilityNeeds: 'Mobility needs',
+    visualImpairment: 'Visual Impairment',
+    isPersonPregnant: 'Is this person pregnant?',
+    expectedDeliveryDate: 'What is their expected date of delivery?',
+    otherPregnancyConsiderations: 'Are there any other considerations',
+    childRemoved: 'Will the child be removed at birth?',
+  }
+
+  yesToPregnancyHealthcareQuestion: boolean = this.answeredYesToPregnancyHealthcareQuestion()
+
+  constructor(
+    private _body: Partial<AccessNeedsFurtherQuestionsBody>,
+    private readonly application: ApprovedPremisesApplication,
+  ) {}
+
+  public set body(value: Partial<AccessNeedsFurtherQuestionsBody>) {
+    if (value.isPersonPregnant === 'yes') {
+      this._body = {
+        ...value,
+        'expectedDeliveryDate-year': value['expectedDeliveryDate-year'] as string,
+        'expectedDeliveryDate-month': value['expectedDeliveryDate-month'] as string,
+        'expectedDeliveryDate-day': value['expectedDeliveryDate-day'] as string,
+        expectedDeliveryDate: DateFormats.dateAndTimeInputsToIsoString(
+          value as ObjectWithDateParts<'expectedDeliveryDate'>,
+          'expectedDeliveryDate',
+        ).expectedDeliveryDate,
+      }
+    }
+  }
+
+  public get body(): AccessNeedsFurtherQuestionsBody {
+    return this._body as AccessNeedsFurtherQuestionsBody
+  }
+
+  previous() {
+    return 'access-needs'
+  }
+
+  next() {
+    return 'covid'
+  }
+
+  answeredYesToPregnancyHealthcareQuestion() {
+    return retrieveOptionalQuestionResponseFromApplicationOrAssessment(
+      this.application,
+      AccessNeeds,
+      'additionalNeeds',
+    ).includes('pregnancy')
+  }
+
+  response() {
+    const response = {
+      [this.questions.wheelchair]: sentenceCase(this.body.needsWheelchair),
+      [this.questions.mobilityNeeds]: this.body.mobilityNeeds,
+      [this.questions.visualImpairment]: this.body.visualImpairment,
```

Comment:
Wondering about these questions - should we conditionally reveal these questions if the relevant additionalNeeds option was set? It seems weird to ask about mobility needs or visual impairments if they weren't previously stated.

Reply:
Yeah good point, I think this would really be better as two seperate pages as the information seems sort of unrelated. Something to talk about with Beth during E2E review